src/ui/views/Approval/components/DefaultBlock.tsx

```diff
@@ -29,6 +29,40 @@ export const DefaultBlock = ({
 }) => {
   const history = useHistory();
 
+  const hexToString = (hex) => {
+    let str = '';
+    for (let i = 0; i < hex.length; i += 2) {
+      str += String.fromCharCode(parseInt(hex.substr(i, 2), 16));
+    }
+    return str;
+  };
+
+  const processItem = (item) => {
+    if (Array.isArray(item)) {
+      return `[${item.map((value) => processItem(value)).join('')}]`;
+    } else if (typeof item === 'object' && item !== null) {
+      if (item.type && item.value !== undefined) {
+        return `{type:${item.type} value:${processItem(item.value)}}`;
+      } else {
+        return `{${Object.entries(item)
+          .map(([key, value]) => `${key}:${processItem(value)}`)
+          .join('')}}`;
+      }
+    } else if (typeof item === 'string' && /^[0-9a-fA-F]+$/.test(item)) {
+      return hexToString(item);
```

Comment:
Why do we need this step ? Could you give an example ?

Reply:
EVM arguments have those hex string needed to decode, flow doesn't seem to have those, I removed these line